packages/@aws-cdk/aws-ec2/lib/vpc-flow-logs.ts

```diff
@@ -0,0 +1,325 @@
+  /**
+   * The name of the FlowLog
+   *
+   * It is not recommended to use an explicit name.
+   *
+   * @default If you don't specify a flowLogName, AWS CloudFormation generates a
+   * unique physical ID and uses that ID for the group name.
+   */
+  readonly flowLogName?: string;
+
+  /**
+   * The type of resource for which to create the flow log
+   */
+  readonly resourceType: FlowLogResourceType;
+}
+
+/**
+ * The base class for a Flow Log
+ *
+ * @experimental
+ */
+abstract class FlowLogBase extends Resource implements IFlowLog {
+  /**
+   * The Id of the VPC Flow Log
+   *
+   * @attribute
+   */
+  public abstract readonly flowLogId: string;
+}
+
+/**
+ * A VPC flow log.
+ * @resource AWS::EC2::FlowLog
+ *
+ * @experimental
+ */
+export class FlowLog extends FlowLogBase {
+  /**
+   * Import a Flow Log by it's Id
+   */
+  public static fromFlowLogId(scope: Construct, id: string, flowLogId: string): IFlowLog {
+    class Import extends FlowLogBase {
+      public flowLogId = flowLogId;
+    }
+
+    return new Import(scope, id);
+  }
+
+  /**
+   * The Id of the VPC Flow Log
+   *
+   * @attribute
+   */
+  public readonly flowLogId: string;
+
+  /**
+   * The S3 bucket to publish flow logs to
+   */
+  public readonly bucket?: s3.IBucket;
+
+  /**
+   * The iam role used to publish logs to CloudWatch
+   */
+  public readonly iamRole?: iam.IRole;
+
+  /**
+   * The CloudWatch Logs LogGroup to publish flow logs to
+   */
+  public readonly logGroup?: logs.ILogGroup;
+
+  constructor(scope: Construct, id: string, props: FlowLogProps) {
+    super(scope, id, {
+      physicalName: props.flowLogName
+    });
+
+    const destination = props.destination || new FlowLogDestinationImpl({
+      logDestinationType: 'cloud-watch-logs'
+    });
+
+    const destinationConfig = destination.toDestination();
+
+    if (destinationConfig.logDestinationType === 'cloud-watch-logs') {
+      this.iamRole = destinationConfig.iamRole || new iam.Role(this, 'IAMRole', {
+        roleName: PhysicalName.GENERATE_IF_NEEDED,
+        assumedBy: new iam.ServicePrincipal('vpc-flow-logs.amazonaws.com')
+      });
+
+      this.logGroup = new logs.LogGroup(this, 'LogGroup', props.logGroupOptions);
+
+      new iam.Policy(this, 'Policy', {
+        roles: [this.iamRole],
+        statements: [
+          new iam.PolicyStatement({
+            actions: [
+              'logs:CreateLogStream',
+              'logs:PutLogEvents',
+              'logs:DescribeLogStreams'
+            ],
+            effect: iam.Effect.ALLOW,
+            resources: [this.logGroup.logGroupArn]
+          }),
+          new iam.PolicyStatement({
+            actions: ['iam:PassRole'],
+            effect: iam.Effect.ALLOW,
+            resources: [this.iamRole.roleArn]
+          })
+        ]
+      });
+    } else {
+      this.bucket = destinationConfig.s3Bucket || new s3.Bucket(this, 'S3Bucket', {
+        encryption: s3.BucketEncryption.UNENCRYPTED
```

Comment:
Might want to put a RETAIN on here as well or you're going to have trouble deleting your stack.

Reply:
updated.